qcore/geo.py

```diff
@@ -1247,3 +1248,40 @@ def rotation_matrix(angle: float) -> np.ndarray:
         The 2x2 rotation matrix.
     """
     return np.array([[np.cos(angle), -np.sin(angle)], [np.sin(angle), np.cos(angle)]])
+
+
+def point_to_segment_distance(
+    p: npt.ArrayLike, q: npt.ArrayLike, r: npt.ArrayLike
+) -> float:
+    """Compute the shortest distance between a point and a line segment.
```

Comment:
Nice simplification! I think it would be good to add a link to the math exchange question that this is based on. It provides lots of details for the non-fluent math users.